### PR TITLE
ENG-14612: Implement cyral_policy_wizards resource

### PR DESCRIPTION
**Pull request by yoursnerdly**
## Description of the change

As title

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [X] Lint rules pass locally
- [X] The code changed/added as part of this pull request has been covered with tests
- [X] All tests related to the changed code pass in development

### Code review

- [X] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [X] Jira issue referenced in commit message and/or PR title

### Testing

- With wizard-id specified, the data source state shows only the requested wizard.
- With no wizard-id specified, the data source state shows the list of all supported wizards.

### CODE REVIEW

**Review by wcmjunior — COMMENTED** [review]
A few suggestions:
1. If the api is `policyWizard` in the singular, please use the same convention for the resource. I remember mentioning it should be in plural, but it was a mistake on my end as I did not check if the API was plural as well.
2. As for code hygiene, I suggest to keep the `policy_wizard` in its own package. Seems like the package structure should be:

```
- policy
   |- set
   |- wizard
```

**Comment by yoursnerdly** [comment]
> 1. If the api is `policyWizard` in the singular, please use the same convention for the resource. I remember mentioning it should be in plural, but it was a mistake on my end as I did not check if the API was plural as well.

Well the [REST path](https://github.com/cyralinc/policy/blob/ae02ad0bbda7683f60bd398e1ffbf2943dd4b9d2/proto/policy/v1/policyset.proto#L35:L47) is `GET /v1/policyWizards` (plural) with optional id at the end. The gRPC methods are called `ReadPolicyWizard` (singular because it pertains to ONE wizard) and `ListPolicyWizards` (plural).

Also given that the data source returns a _list_ of wizards, I'd suggest retaining the plural.

**Comment by yoursnerdly** [comment]
> ```
> - policy
>    |- set
>    |- wizard
> ```

Thanks for the suggestion, this is incorporated in the latest commit.

**Review by gengdahlCyral — APPROVED** [review]
LGTM

File: cyral/internal/policy/set/constants.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package policyset
+
+const (
+	policySetResourceName   = "cyral_policy_set"
+	policySetDataSourceName = policySetResourceName
```
Comment (wcmjunior) [inline cyral/internal/policy/set/constants.go:5]:
```suggestion
	resourceName   = "cyral_policy_set"
	dataSourceName = resourceName
```

File: cyral/internal/policy/set/datasource.go [hunk 2]
```diff
@@ -7,16 +7,16 @@ import (
 	"github.com/cyralinc/terraform-provider-cyral/cyral/core/types/resourcetype"
 )
 
-var dsContextHandler = core.ContextHandler{
-	ResourceName: dataSourceName,
+var policySetDataSourceContextHandler = core.ContextHandler{
```
Comment (wcmjunior) [inline cyral/internal/policy/set/datasource.go:10]:
```suggestion
var dsContextHandler = core.ContextHandler{
```

File: cyral/internal/policy/set/datasource.go [hunk 3]
```diff
@@ -7,16 +7,16 @@ import (
...
+	ResourceName: policySetDataSourceName,
 	ResourceType: resourcetype.DataSource,
 	Read:         readPolicySet,
 }
 
-func dataSourceSchema() *schema.Resource {
+func policySetDataSourceSchema() *schema.Resource {
```
Comment (wcmjunior) [inline cyral/internal/policy/set/datasource.go:16]:
```suggestion
func dataSourceSchema() *schema.Resource {
```

File: cyral/internal/policy/set/datasource.go [hunk 4]
```diff
@@ -7,16 +7,16 @@ import (
...
 	return &schema.Resource{
 		Description: "This data source provides information about a policy set.",
-		ReadContext: dsContextHandler.ReadContext,
+		ReadContext: policySetDataSourceContextHandler.ReadContext,
```
Comment (wcmjunior) [inline cyral/internal/policy/set/datasource.go:19]:
```suggestion
		ReadContext: dsContextHandler.ReadContext,
```

File: cyral/internal/policy/set/model.go [hunk 5]
```diff
@@ -47,8 +47,8 @@ func scopeToMap(s *msg.Scope) []map[string]interface{} {
 	}
 }
 
-// updateSchema writes the policy set data to the schema
-func updateSchema(ps *msg.PolicySet, d *schema.ResourceData) error {
+// updatePolicySetSchema writes the policy set data to the schema
+func updatePolicySetSchema(ps *msg.PolicySet, d *schema.ResourceData) error {
```
Comment (wcmjunior) [inline cyral/internal/policy/set/model.go:51]:
```suggestion
func updateSchema(ps *msg.PolicySet, d *schema.ResourceData) error {
```

File: cyral/internal/policy/set/resource.go [hunk 6]
```diff
@@ -9,22 +9,22 @@ import (
 	"github.com/cyralinc/terraform-provider-cyral/cyral/core/types/resourcetype"
 )
 
-var resourceContextHandler = core.ContextHandler{
-	ResourceName: resourceName,
+var policySetResourceContextHandler = core.ContextHandler{
```
Comment (wcmjunior) [inline cyral/internal/policy/set/resource.go:12]:
```suggestion
var resourceContextHandler = core.ContextHandler{
```

File: cyral/internal/policy/set/resource.go [hunk 7]
```diff
@@ -9,22 +9,22 @@ import (
...
+	ResourceName: policySetResourceName,
```
Comment (wcmjunior) [inline cyral/internal/policy/set/resource.go:13]:
```suggestion
	ResourceName: resourceName,
```

File: cyral/internal/policy/set/resource.go [hunk 8]
```diff
@@ -9,22 +9,22 @@ import (
...
 	ResourceType: resourcetype.Resource,
 	Create:       createPolicySet,
 	Read:         readPolicySet,
 	Update:       updatePolicySet,
 	Delete:       deletePolicySet,
 }
 
-func resourceSchema() *schema.Resource {
+func policySetResourceSchema() *schema.Resource {
```
Comment (wcmjunior) [inline cyral/internal/policy/set/resource.go:21]:
```suggestion
func resourceSchema() *schema.Resource {
```

**Review by wcmjunior — COMMENTED** [review]
The convention documented [here](https://github.com/cyralinc/terraform-provider-cyral/blob/main/cyral/core/README.md) is used throughout the whole codebase makes it faster to anyone reading it to understand what is going on. Maybe the docs are not good enough to ask to follow the same naming conventions, but if you check all the other resources they will look the very same as the changes I propose in the review.

As we use clear and descriptive package names like `policy.set` and `policy.wizard`, we can use the same standards to make it easier to anyone used to read any of our resources and data sources. We also end up avoid [Smurf naming convention](https://devcards.io/smurf-naming-convention).

**Comment by yoursnerdly** [comment]
> The convention documented [here](https://github.com/cyralinc/terraform-provider-cyral/blob/main/cyral/core/README.md) used throughout the whole codebase makes it faster to anyone reading it to understand what is going on. Maybe the docs are not good enough to ask to follow the same naming conventions, but if you check all the other resources they will look the very same as the changes I propose in the review.
> 
> As we use clear and descriptive package names like `policy.set` and `policy.wizard`, we can use the same standards to make it easier to anyone used to read any of our resources and data sources. We also end up avoid [Smurf naming convention](https://devcards.io/smurf-naming-convention).

Yeah, this happened due to refactoring out into separate packages for policy set and wizard - I had to use different names when they were in the same package. Will fix.

**Review by wcmjunior — APPROVED** [review]

**Pull request merged**